src/parcsr_ls/par_relax.c

```diff
@@ -1409,3 +1466,150 @@ hypre_BoomerAMGRelaxKaczmarz( hypre_ParCSRMatrix *A,
    return hypre_error_flag;
 }
 
+HYPRE_Int
+hypre_BoomerAMGRelax11TwoStageGaussSeidelHost( hypre_ParCSRMatrix *A,
+                                               hypre_ParVector    *f,
+                                               HYPRE_Real          relax_weight,
+                                               HYPRE_Real          omega,
+                                               hypre_ParVector    *u,
+                                               hypre_ParVector    *Vtemp )
+{
+   hypre_CSRMatrix *A_diag      = hypre_ParCSRMatrixDiag(A);
+   HYPRE_Int        num_rows    = hypre_CSRMatrixNumRows(A_diag);
+   HYPRE_Real      *A_diag_data = hypre_CSRMatrixData(A_diag);
+   HYPRE_Int       *A_diag_i    = hypre_CSRMatrixI(A_diag);
+   HYPRE_Int       *A_diag_j    = hypre_CSRMatrixJ(A_diag);
+   hypre_Vector    *Vtemp_local = hypre_ParVectorLocalVector(Vtemp);
+   HYPRE_Complex   *Vtemp_data  = hypre_VectorData(Vtemp_local);
+   hypre_Vector    *u_local     = hypre_ParVectorLocalVector(u);
+   HYPRE_Complex   *u_data      = hypre_VectorData(u_local);
+   HYPRE_Int        i, jj, ii;
+
+
+   /* Need to check that EVERY diagonal is nonzero first. If any are, throw exception */
+   for (i = 0; i < num_rows; i++)
+   {
+      if (A_diag_data[A_diag_i[i]] == 0.0)
+      {
+         hypre_error_in_arg(1);
```

Comment:
@PaulMullowney Please check if this error handling makes sense. Thanks!

Reply:
I think this is fine.